src/widget/form/settings/generalform.cpp

```diff
@@ -210,6 +211,16 @@ void GeneralForm::onTranslationUpdated()
 {
     Settings::getInstance().setTranslation(locales[bodyUI->transComboBox->currentIndex()]);
     Widget::getInstance()->setTranslation();
+    Settings::getInstance().save();
+    QMessageBox::StandardButton reply;
+      reply = QMessageBox::question(0,
+                                    tr("Do you want to restart now?"),
+                                    tr("You must qTox restart to change the language"),
+                                    QMessageBox::Yes|QMessageBox::No);
+      if (reply == QMessageBox::Yes) {
+          QProcess::startDetached(QApplication::applicationFilePath());
+          exit(12);
```

Comment:
Yes, it works with any return code here.

The problem is not the return code. It's probably just the new instance racing with the current one. If it starts before we exit, it'll just maximize the current running instance.
So we need to release the IPC's hold on the shared memory before starting the new process.